src/internal/RSocketConnectionManager.cpp

```diff
@@ -87,8 +88,20 @@ void RSocketConnectionManager::manageConnection(
       *this, socket, eventBase);
   newNetworkStats->inner = std::move(socket->networkStats());
   socket->networkStats() = std::move(newNetworkStats);
+  sockets_.lock()->insert({std::move(socket), {eventBase, resumeToken}});
```

Comment:
I understand the intention to keep the application code as simple as possible and so storing the ResumeIdentificationToken here. My concern is that the library is now in the business of storing it, looking up state machine by RID, dealing with problems with duplicity or empty values, ...
I was preferring in pushing this problem into the application code.